docs/references/astro/endpoints.mdx

```diff
@@ -0,0 +1,90 @@
+---
+title: Endpoints
+description: Learn how to use Clerk with your Endpoints.
+---
+
+# Endpoints
+
+Clerk provides helpers that allow you to protect your [Endpoints](https://docs.astro.build/en/guides/endpoints/#server-endpoints-api-routes), fetch the current user, and interact with the Clerk backend API.
+
+## Protect your Endpoints
+
+If you aren't protecting your Endpoints using [`clerkMiddleware()`](/docs/references/astro/clerk-middleware), you can protect your Endpoints by using the `auth()` local and check for the `userId` value:
+
+
+```ts filename="src/pages/api/route.ts"
+export async function GET({ locals }) {
+  const { userId } = locals.auth();
+
+  if (!userId) {
+    return new Response('Error: No signed in user', { status: 401 })
+  }
+
+  // Add your Endpoint logic here
+
+  return new Response(JSON.stringify({ userId }));
+}
+```
+
+## Retrieve data from external sources
+
+Clerk provides integrations with a number of popular databases.
+
+The following example demonstrates how to use [`auth().getToken()`](/docs/references/nextjs/auth-object#get-token) to retrieve a token from a JWT template and use it to fetch data from the external source.
+
+```ts filename="src/pages/api/route.ts"
+export async function GET({ locals }) {
+  const { userId, getToken } = locals.auth();
+
+  if (!userId) {
+    return new Response('Unauthorized', { status: 401 })
+  }
+
+  const token = await getToken({ template: "supabase" });
+
+  // Fetch data from Supabase and return it.
+  const data = { supabaseData: 'Hello World' };
+
+  return new Response(JSON.stringify(data));
+}
+```
+
+## Retrieve the current user
+
+In some cases, you might need the current user in your Endpoint. Clerk provides an asynchronous local called [`currentUser()`](/docs/references/nextjs/current-user) to retrieve the current [`Backend User`](/docs/references/backend/types/backend-user) object.
+
+```ts filename="src/pages/api/route.ts"
+export async function GET({ locals }) {
+  const user = await locals.currentUser();
+
+  if(!user){
+    return new Response("Unauthorized", { status: 401 });
+  }
+
+  return new Response(JSON.stringify({ user }));
+}
+```
+
+## Interact with Clerk's Backend API
+
+The [JavaScript Backend SDK](/docs/references/backend/overview) exposes [Clerk's backend API](https://clerk.com/docs/reference/backend-api) resources and low-level authentication utilities for JavaScript environments.
+
+`clerkClient` exposes an instance of the JavaScript Backend SDK for use in server environments.
+
+```ts filename="src/pages/api/route.ts"
+import { clerkClient } from '@clerk/astro/server';
+
+export async function POST(context) {
+  const { userId } = context.locals.auth();
+
+  if(!userId){
+    return new Response("Unauthorized", { status: 401 });
+  }
+
+  const params = { firstName: 'John', lastName: 'Wick' };
```

Comment:
Unless we use this already, I consider this
```suggestion
  const params = { firstName: 'Clerk', lastName: 'Cookie' };
```

Reply:
We use this name in Nextjs example if I remember correctly but approving your suggestion since it sounds more professional :D